### PR TITLE
Scrub and collect cilium-etcd-secrets as part of sysdump

### PR DESCRIPTION
**Pull request by abhide15**
Signed-off-by: Amey Bhide <amey@covalent.io>

### CODE REVIEW

File: cluster-diagnosis/sysdumpcollector.py [hunk 1]
```diff
@@ -200,6 +202,27 @@ def collect_cilium_configmap(self):
             log.info("collected cilium configmap yaml file: {}".format(
                 configmap_file_name))
 
+    def collect_cilium_secret(self):
+        secret_file_name = "cilium-etcd-secrets-{}.json".format(
+            utils.get_current_time())
+        cmd = "kubectl get secret cilium-etcd-secrets -n kube-system -ojson"
```
Comment (ashwinp) [inline cluster-diagnosis/sysdumpcollector.py]: for consistency with other code... `-o json`

File: cluster-diagnosis/sysdumpcollector.py [hunk 2]
```diff
@@ -200,6 +202,27 @@ def collect_cilium_configmap(self):
...
+            data = {}
+            for key, value in output.get('data').iteritems():
+                data[key] = "XXXXX"
+            output['data'] = data
+            with open(
+                os.path.join(self.sysdump_dir_name, secret_file_name), 'w') as fp:
+                fp.write(json.dumps(output))
+        except subprocess.CalledProcessError as exc:
+            if exc.returncode != 0:
+                log.error("Error: {}. Unable to get cilium secret json")
```
Comment (ashwinp) [inline cluster-diagnosis/sysdumpcollector.py]: print the error. `Error: {}` param for {} is missing.

Comment (ashwinp) [reply cluster-diagnosis/sysdumpcollector.py]: Change the log to "Unable to retrieve and redact cilium secret."

File: cluster-diagnosis/sysdumpcollector.py (hunk 1 above)
Comment (abhide15) [reply cluster-diagnosis/sysdumpcollector.py]: Done

File: cluster-diagnosis/sysdumpcollector.py (hunk 2 above)
Comment (abhide15) [reply cluster-diagnosis/sysdumpcollector.py]: Done

File: cluster-diagnosis/sysdumpcollector.py [hunk 3]
```diff
@@ -195,11 +198,34 @@ def collect_cilium_configmap(self):
...
 
+    def collect_cilium_secret(self):
+        secret_file_name = "cilium-etcd-secrets-{}.json".format(
+            utils.get_current_time())
+        cmd = "kubectl get secret cilium-etcd-secrets -n kube-system -o json"
+        try:
+            output = json.loads(subprocess.check_output(cmd, shell=True))
+            data = {}
+            for key, value in output.get('data').iteritems():
+                data[key] = "XXXXX"
```
Comment (ashwinp) [inline cluster-diagnosis/sysdumpcollector.py]: nitpick: Change XXXX to **** . That's a more common form to represent redacted data.

**Pull request merged**